### PR TITLE
[IMP] runbot_travis2docker: Start ssh service

### PR DESCRIPTION
**Pull request by moylop260**
- [x] Start ssh service
- [x] Add committer's github keys to `/home/odoo/.ssh/authorized_keys` (With `odoo:odoo` owner too)
- [x] Add tests to check
   - [x] if the file `/home/odoo/.ssh/authorized_keys` was updated from container
   - [x] if the field `build.docker_executed_commands` was updated
   - [x] if the `/etc/init.d/ssh status` is running

### CODE REVIEW

**Comment by moylop260** [comment]
@JesusZapata 
Could you help me to finish this PR, please? (Check the main description)

**Comment by JesusZapata** [comment]
@moylop260 
Only Just need to calculate the github user
You know how to find this information starting of repository?

**Comment by moylop260** [comment]
The json returned from [_get_pull_info](https://github.com/odoo/odoo-extra/blob/851320c0175/runbot/runbot.py#L504) method has user info
  - E.g. https://api.github.com/repos/odoo/odoo/pulls/1

For stable branches `refs/head/10.0` you could use a github api too (see the documentation)

For get the keys you can use: https://github.com/USER.keys
  - E.g. https://github.com/jesuszapata.keys


Note: You consider this change for gitlab too

**Comment by JesusZapata** [comment]
@moylop260 
Thanks for you help.
I made the test for this funcionality

File: runbot_travis2docker/models/runbot_build.py [hunk 1]
```diff
@@ -355,3 +359,53 @@ def _local_cleanup(self, cr, uid, ids, context=None):
             if build.docker_container:
                 build.docker_rm_container()
                 build.docker_rm_image()
+
+    def schedule(self, cr, uid, ids, context=None):
+        res = super(RunbotBuild, self).schedule(cr, uid, ids, context=context)
+        for build in self.browse(cr, uid, ids, context=context):
+            if (build.state == 'running' and build.job == 'job_30_run' and
+                    not build.docker_executed_commands):
+                session = requests.Session()
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_build.py]: What about use `repo.github` method?

File: runbot_travis2docker/models/runbot_build.py [hunk 2]
```diff
@@ -355,3 +359,53 @@ def _local_cleanup(self, cr, uid, ids, context=None):
...
+                    if 'done.' not in output:
+                        _logger.exception('Fail to start ssh server: %s',
+                                          output)
+                    else:
+                        _logger.info(output)
+                    cmd = ["docker", "exec", "--user=odoo",
+                           build.docker_container, "curl",
+                           "https://github.com/%s.keys" %
+                           json['author']['login'], "-o",
+                           "/home/odoo/.ssh/authorized_keys"]
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_build.py]: The original ones shouldn't be overwritten

File: runbot_travis2docker/models/runbot_build.py [hunk 3]
```diff
@@ -355,3 +359,53 @@ def _local_cleanup(self, cr, uid, ids, context=None):
...
+                    _logger.info('Copy keys of github : ' + ' '.join(cmd))
+                    subprocess.call(cmd)
+                    cmd = ["docker", "exec", "--user=odoo",
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_build.py]: This validation should be moved to tests section
IMHO here we are running `docker-exec` commands unnecessarily

File: runbot_travis2docker/tests/test_runbot_build.py [hunk 4]
```diff
@@ -78,6 +78,38 @@ def test_20_jobs_pr(self):
         _logger = logging.getLogger(__name__ + '.def test_20_jobs_pr')
         self.run_jobs('refs/pull/1')
 
+    def test_30_ssh_server(self):
+        self.repo.update()
+        self.repo.killall()
```
Comment (moylop260) [inline runbot_travis2docker/tests/test_runbot_build.py:83]: These tests are too much expensive
Maybe we could re-use the original method to add custom validations for ssh keys
(I know we are breaking tests' standards)

**Review by moylop260 — COMMENTED** [review]
@JesusZapata Could you check if is good for you my last commit?

**Comment by JesusZapata** [comment]
@moylop260 
I reviewed your change and all it is okay
Thanks for yours commits

**Pull request merged**